### ISSUE TITLE
SuperPuTTY corrupts layout file after Windows restarts, then "Confirm exit" is enabled.

### ISSUE THREAD

**Issue by vityank** [jimradford/superputty#789]
**Describe the bug**
SuperPuTTY corrupts layout file after Windows restarts, then "Confirm exit" is enabled.

**To Reproduce**
SuperPuTTY - "Confirm exit" option enabled.
Then Windows updates with automatic enforcing of reboot applied and system restarted(While SuperPuTTY blocked it for a while). Note - User is not in front of PC during this time,
AutoRestoreLayout.XML get's overriden with 'default' version - causing the layout of tabs to be completely lost.
Looks like SuperPuTTY performs it's OnClose callbacks but the objects dumped already partially or completely destructed by this time. 
The issue can't be reproduced by simple 'kills' of process and/or it's tree.

**Desktop (please complete the following information):**
 - Windows 10 1809 and 1607

**Comment by vityank** [comment]
Found a way to reproduce and explanation of that exactly happens.
OS starting to kill processes.
SuperPuTTY waits with Close approval window.
In parallel OS kills PuTTY/KiTTY processes - The tabs in SuperPuTTY disappear one by one for each such process.
Finally internal timeout in SuperPuTTY is reached and it exists with own code(Not killed), and saves the automatic layout.

To reproduce:
Open SuperPuTTY with your layout.
Start killing PuTTY/KiTTY processes in Task Manager. Check how SuperPuTTY reacts.
I still not decided what a best wait to fix it.

**Comment by jimradford** [comment]
@vityank probably the most logical way to handle it is to subscribe to the shutdown / end session event and handle it prior to windows killing processes, e.g. graceful exit. GREAT WORK btw.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.win32.systemevents.sessionending?redirectedfrom=MSDN&view=netframework-4.8

**Comment by vityank** [comment]
I did similar tot his but using other strategy to not touch the SuperPuTTY's scheme of interaction with it's child processes.
I just added a check there auto-layout save occurs for exit reason. For OS-Shutdown or TaskMan kill I just skip the save.
Work like a charm in testing.
100% bug reproduce:
Open SuperPuTTY with layout
Reboot using from cmd: shutdown -r -f
Wait 10s fro trigerring.
After reboot SuperPuTTY will have empty layout.

Just added PR with above-described algo.
